tools/bin/commands/release

```diff
@@ -0,0 +1,149 @@
+
+REPO_ANNOUNCEMENT_PREFIX="OpenedStagingProfile"
+
+release::description() {
+    echo "Perform a release"
+}
+
+release::usage() {
+    cat - <<EOT
+    --dry-run                 Dry run. This will stage and drop a release
+    --release-version         The release version
+    --dev-version             The development version
+    --profiles                Additional profiles to specify
+    --settings-xml            Path to a custom settings.xml to use for the reelase
+
+    --sonatype-username       The sonatype username
+    --sonatype-password       The sonatype password
+
+    --docker-registry         The docker registry
+    --docker-server-id        The docker server id (as defined in settings.xml)
+    --docker-username         The docker username
+    --docker-email            The docker email
+    --docker-password         The docker password
+
+
+EOT
+}
+
+release::run() {
+    source "$(basedir)/commands/build"
+    local dry_run=$(hasflag --dry-run)
+    local release_version=$(readopt --release-version)
+    local dev_version=$(readopt --dev-version)
+    local maven_opts=""
+
+    if [ -z "${release_version}" ]; then
+        echo "Please specify --release-version"
+        exit 1
+    fi
+
+    if [ -z "${dev_version}" ]; then
+        echo "Please specify --dev-version"
+        exit 1
+    fi
+
+    local profiles=$(readopt --profiles)
+    if [ -n "${profiles}" ]; then
+        maven_opts="$maven_opts -P${profiles}"
+    else
+        maven_opts="-Prelease"
+    fi
+
+    local settings_xml=$(readopt --settings-xml)
+    if [ -n "${settings_xml}" ]; then
+        maven_opts="$maven_opts -s $settings_xml"
+    fi
+
+    echo "Creating new staging repository..."
+    local repo_id=$(open_staging_repository "$maven_opts")
+    echo "Opened staging repository: $repo_id"
+    trap "do_cleanup \"$repo_id\" \"none\" \"$maven_opts\"" EXIT
+
+    do_release "$release_version" "$dev_version" "$maven_opts"
+    local tag=$(find_tag)
+    trap "do_cleanup \"$repo_id\" \"none\" \"$maven_opts\"" EXIT
+
+    echo "Closing staging repository: $repo_id"
+    close_staging_repository "$repo_id" "$maven_opts"
+
+    if [ -n "$dry_run" ]; then
+        echo "This is a dry run ..."
+        echo "Droping tag $tag"
+        git tag -d $tag
+        echo "Droping staging repository: $repo_id"
+        drop_staging_repository "$repo_id" "$maven_opts"
+    else
+        echo "Releasing staging repository: $repo_id"
+        trap "echo Done!" EXIT
+        exit
+        release_staging_repository "$repo_id" "$maven_opts"
+        git push origin master
+        git push orign $tag
+    fi
+}
+
+# ======================================================
+# Staging / Release Helper Functions
+
+latest_release() {
+    git log | grep "prepare release $artifactId-" | head -n 1 | awk -F "$artifactId-" '{print \$2}'
```

Comment:
now that we have a single version is this same as `git tag|tail -1`, also I can't see this being used here, perhaps it can be removed?

Reply:
Yes, this needs to go away